.github/workflows/ci.yml

```diff
@@ -0,0 +1,72 @@
+name: CI Checks
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Build Docker Image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: get short SHA
+        id: sha
+        run: |
+          echo ::set-output name=sha_short::$(git rev-parse --short=7 ${{ github.sha }})
+      - name: Docker meta
+        id: docker_meta
+        uses: crazy-max/ghaction-docker-meta@v1
+        with:
+          images: quay.io/dannymassa/testrepo
+          # Add custom labels that aren't generated by docker-meta action
+          label-custom: |
+            org.opencontainers.image.authors=AT&T Services INC
+            org.opencontainers.image.description=Tekton/Gerrit Connection Microservice
+            org.opencontainers.image.documentation=https://github.com/att-comdev/jarvis-connector
+            org.opencontainers.image.vendor=AT&T Services INC
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build Without Pushing
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          tags: |
+            quay.io/attcomdev/jarvis-connector:${{ steps.sha.outputs.sha_short }}
+          labels: ${{ steps.docker_meta.outputs.labels }}
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.33
+        env:
+          ACTIONS_ALLOW_UNSECURE_COMMANDS: 'true'
+  test:
+    name: Build & Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Go 1.13
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.13
+      - name: Get dependencies
+        run: |
+          go get -v -t -d ./...
+          if [ -f Gopkg.toml ]; then
+              curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+              dep ensure
+          fi
+      - name: Build
+        run: go build -v ./...
+      - name: Test
+        run: go test -v ./...

```

Comment:
This we will also want controlled in the dockerfile, you should be ab le to add a line verify similar to what you have here, again, not wanting to rely entirely on github actions if this image is built elsewhere.